egs/wsj/s5/steps/nnet3/chain2/process_egs.sh

```diff
@@ -0,0 +1,171 @@
+  echo "Main options (for others, see top of script file)"
+  echo "  --config <config-file>                           # config file containing options (alternative to this"
+  echo "                                                   # command line)"
+  echo "  --cmd (utils/run.pl;utils/queue.pl <queue opts>) # how to run jobs."
+  echo "  --chunks-per-group <n;4>                           # Number of chunks (preferentially, from a single speaker"
+  echo "                                                   # to combine into each example.  This grouping of"
+  echo "                                                   # egs is part of the 'chaina' framework; the adaptation"
+  echo "                                                   # parameters will be estimated from these groups."
+  echo "  --num-repeats <n;2>                              # Number of times we group the same chunks into different"
+  echo "                                                   # groups.  For now only the values 1 and 2 are"
+  echo "                                                   # recommended, due to the very simple way we choose"
+  echo "                                                   # the groups (it's consecutive)."
+  echo "  --nj       <num-jobs;5>                          # Number of jobs to run in parallel.  Usually quite a"
+  echo "                                                   # small number, as we'll be limited by disk access"
+  echo "                                                   # speed."
+  echo "  --compress <bool;true>                           # True if you want the egs to be compressed"
+  echo "                                                   # (e.g. you may set to false for debugging purposes, to"
+  echo "                                                   # check that the compression is not hurting)."
+  echo "  --num-heldout-egs <n;200>                        # Number of egs to put in train_subset.scp and heldout_subset.scp."
+  echo "                                                   # These will be used for diagnostics.  Note: this number is"
+  echo "                                                   # the number of  grouped egs, after merging --chunks-per-group"
+  echo "                                                   # chunks into a single eg."
+  echo "                                                   # ... may be a comma separated list, but we advise a single"
+  echo "                                                   #  number in most cases, due to interaction with the need "
+  echo "                                                   # to group egs from the same speaker into groups."
+  echo "  --stage <stage|0>                                # Used to run this script from somewhere in"
+  echo "                                                   # the middle."
+  exit 1;
+fi
+
+raw_egs_dir=$1
+dir=$2
+
+# die on error or undefined variable.
+set -e -u
+
+if ! steps/chain/validate_raw_egs.sh $raw_egs_dir; then
```

Comment:
Should this be `steps/nnet3/chain2/validate_raw_egs.sh`?

Reply:
yes! I have fixed this and will be pushing the changes with the BABEL recipe soon. I should probably run the tests for mini_librespeech again.

Reply:
this is fixed in the latest commit.